src/pages/api/auth/[...nextauth].ts

```diff
@@ -39,90 +40,108 @@ export const adapter = SequelizeAdapter(sequelize, {
   models: { User: db.User },
 });
 
-export const authOptions: NextAuthOptions = {
-  adapter,
-
-  session: {
-    strategy: "jwt",
-  },
-
-  providers: [
-    // @ts-expect-error
-    {
-      id: 'sendgrid',
-      type: 'email',
-      maxAge: tokenMaxAgeInMins * 60, // For verification token expiry
-      sendVerificationRequest,
-      generateVerificationToken: generateShortLivedToken,
+export function authOptions(req?: NextApiRequest): NextAuthOptions {
+  return {
+    adapter,
+  
+    session: {
+      strategy: "jwt",
     },
-    WeChatProvider({
-      id: "wechat-qr",
-      name: "微信扫码登陆",
-      checks: ["none"],
-      clientId: process.env.AUTH_WECHAT_QR_APP_ID!,
-      clientSecret: process.env.AUTH_WECHAT_QR_APP_SECRET!,
-      platformType: "WebsiteApp",
-    }),
-    WeChatProvider({
-      id: "wechat",
-      name: "微信内登陆",
-      clientId: process.env.AUTH_WECHAT_APP_ID!,
-      clientSecret: process.env.AUTH_WECHAT_APP_SECRET!,
-      platformType: "OfficialAccount",
-    })
-  ],
-
-  pages: {
-    signIn: '/auth/login',
-    // The login page respects the `?error=` URL param.
-    error: '/auth/login',
-  },
-
-  callbacks: {
-    async jwt({ token, trigger, session }) {
-      // https://next-auth.js.org/getting-started/client#updating-the-session
-      if (trigger == "update" && session) {
-        const impersonate = (session as ImpersonationRequest).impersonate;
-        if (impersonate === null) {
-          delete token[impersonateTokenKey];
-        } else {
-          const me = await db.User.findByPk(token.sub);
-          invariant(me);
-          if (!isPermitted(me.roles, "UserManager")) {
-            throw noPermissionError("用户", impersonate);
+  
+    providers: [
+      // @ts-expect-error
+      {
+        id: 'sendgrid',
+        type: 'email',
+        maxAge: tokenMaxAgeInMins * 60, // For verification token expiry
+        sendVerificationRequest,
+        generateVerificationToken: generateShortLivedToken,
+      },
+      WeChatProvider({
+        id: "wechat-qr",
+        name: "微信扫码登陆",
+        checks: ["none"],
```

Comment:
Do we still need this line here?

Reply:
// The "checks: ['none']" is necessary because the current QR code login does not follow the full NA auth flow, 
        // and the backend does not generate a state. Removing this would cause the check to fail, preventing login.